### ISSUE TITLE
mypyc `missing terminating ' character` build error

### ISSUE THREAD

**Issue by pgjones** [python/mypy#7622]
When using mypyc to build code of this form,
```python
headers: List[Tuple[bytes, bytes]]
for name, value in headers:
    if name == b"host":
        write(b"%s: %s\r\n" % (name, value))
for name, value in headers:
    if name != b"host": 
        write(b"%s: %s\r\n" % (name, value))
```
an error is reported,
```
build/__native.c:42559:25: error: missing terminating ' character [-Werror,-Winvalid-pp-token]
        PyObject *cpy_r_name';
```
When I switch the second for-loops variable names to be `name_` and `value_` it builds successfully. 

(I think the `if` clause and `write` call are irrelevant to this bug, but I'm unsure).

**Comment by msullivan** [comment]
Good catch. I think you must have `allow_redefinition` turned on, right? It looks like that generates internal variable names with single quotes at the end (serving as "primes" from mathematical notation). I've opened a bug on the mypyc tracker, so I'll be closing this one.

Thanks!
